include/llbuild/Basic/Tracing.h

```diff
@@ -15,57 +15,99 @@
 
 #include "llvm/ADT/StringRef.h"
 
-// Allow vendored llbuild installations to provide custom tracing entry points
-#if __has_include("TracingMacros.h")
-#include "TracingMacros.h"
-#endif
+// os_signpost is included in mac OS 10.14, if that header is not available, we don't trace at all.
+#if __has_include(<os/signpost.h>)
+#include <os/signpost.h>
+
+/// Returns the singleton instance of os_log_t to use with os_log and os_signpost API.
+API_AVAILABLE(macos(10.12), ios(10.0), watchos(3.0), tvos(10.0))
+static os_log_t getLog() {
+  static os_log_t generalLog;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    generalLog = os_log_create("com.apple.dt.XCBuild", "llbuild");
+  });
+  return generalLog;
+}
 
-#ifndef LLBUILD_TRACE_POINT
-#define LLBUILD_TRACE_POINT(kind, arg1, arg2, arg3, arg4) \
-    { (void)(kind); (void)(arg1); (void)(arg2); (void)(arg3); (void)(arg4); }
-#endif
+/// Returns a singleton id to use for all signpost calls during the lifetime of the program.
+API_AVAILABLE(macos(10.14), ios(12.0), tvos(12.0), watchos(5.0))
+static os_signpost_id_t signpost_id() {
```

Comment:
Why not use `OS_SIGNPOST_ID_EXCLUSIVE` here?

Reply:
We can change that in the future, it doesn't make a difference semantically since the only requirement is uniqueness. But it would be better though!